test/expected_comparators/latest/frontend/src/tests/testclient/index.ts

```diff
@@ -0,0 +1,247 @@
+  on(event: AvailableEvents.subscribe, listener: (e: SubEvent) => void): this;
+  on(event: AvailableEvents.unsubscribe, listener: (e: SubEvent) => void): this;
+  on(event: AvailableEvents.yield, listener:  () => void): this;
+}
+export class NatsAsyncApiTestClient extends events.EventEmitter{
+  public jsonClient?: Client;
+  public stringClient?: Client;
+  public binaryClient?: Client;
+  public options?: NatsConnectionOptions;
+
+	/**
+	 *
+	 * @param options options to use, payload is omitted if sat in the AsyncAPI document.
+	 */
+  constructor() {
+    super();
+  }
+
+  /**
+   * Try to connect to the NATS server with the different payloads.
+   */
+  async connect(options : NatsConnectionOptions){
+		this.options = this.setDefaultOptions(options);
+    try{
+      this.options.payload = Payload.JSON;
+      this.jsonClient = await connect(this.options);
+      this.chainEvents(this.jsonClient);
+    }catch(e){
+      this.emit(AvailableEvents.connectionError, e)
+    }
+  }
+
+
+  /**
+   * Disconnect all clients from the server
+   */
+  async disconnect(){
+    this.jsonClient!.close()
+  }
+  
+  private chainEvents(ns: Client){
+      ns.on('permissionError', (e: NatsError) => {
+        this.emit(AvailableEvents.permissionError, e)
+      });
+      ns.on('close', (e: NatsError) => {
+        this.emit(AvailableEvents.close, e)
+      });
+      ns.on('connect', (connection: Client, serverURL: string, info: ServerInfo) => {
+        this.emit(AvailableEvents.connect, connection, serverURL, info)
+      });
+      ns.on('connecting', (serverURL: NatsError) => {
+        this.emit(AvailableEvents.connecting, serverURL)
+      });
+      ns.on('disconnect', (serverURL: string) => {
+        this.emit(AvailableEvents.disconnect, serverURL)
+      });
+      ns.on('error', (e: NatsError) => {
+        this.emit(AvailableEvents.error, e)
+      });
+      ns.on('pingcount', () => {
+        this.emit(AvailableEvents.pingcount)
+      });
+      ns.on('pingtimer', () => {
+        this.emit(AvailableEvents.pingtimer)
+      });
+      ns.on('reconnect', (connection: Client, serverURL: string, info: ServerInfo) => {
+        this.emit(AvailableEvents.reconnect, connection, serverURL, info)
+      });
+      ns.on('reconnecting', (serverURL: string) => {
+        this.emit(AvailableEvents.reconnecting, serverURL)
+      });
+      ns.on('serversChanged', (e: ServersChangedEvent) => {
+        this.emit(AvailableEvents.serversChanged, e)
+      });
+      ns.on('subscribe', (e: SubEvent) => {
+        this.emit(AvailableEvents.subscribe, e)
+      });
+      ns.on('unsubscribe', (e: SubEvent) => {
+        this.emit(AvailableEvents.unsubscribe, e)
+      });
+      ns.on('yield', () => {
+        this.emit(AvailableEvents.yield)
+      });
+  }
+  /**
+   * Try to connect to the NATS server with nkey authentication
+   */
+  async connectWithNkey(options : NatsConnectionOptions, publicNkey: string, seed: string){
```

Comment:
I'd suggest adding a connect method with a credentials file - it's a popular way to connect. 

e.g. usage is `await connect({url: 'tls://localhost:4222', nkeyCreds: "/tmp/seed.txt" });`

Reply:
Great catch and good question! I am still wondering if I should add these connection wrappers for simplification or just provide the default `connect` function and let the developer handle how the client should be connected.

Or if it should provide the wrappers (maybe dependent on the defined [security schemas](https://www.asyncapi.com/docs/specifications/2.0.0/#securitySchemeObject) to ease the interaction such as `connectWithNkey`. 

Any suggestions on what would be the prefered way? 🤔

Reply:
The three most common ways to connect are plain url (no auth), user/pass, and credentials.  I'll leave it up to you to decide what is best for users in terms of API footprint but for ease of use I'd suggest APIs for plain url and credentials.  Can users still establish connections on their own?  For new projects we're recommending using NATS creds to connect.

Reply:
Okay, i'll add url (no auth), user/pass, and credentials as build in connection functions. And yes people can still connect using their own custom connection options in any way they want 👍

Reply:
Thanks - I appreciate it!  Those should cover 99% of NATS use cases.